### PR TITLE
One final fix + cut 0.0.103

### PR DESCRIPTION
**Pull request by TheBlueMatt**

### CODE REVIEW

**Review by valentinewallace — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,64 @@
...
+   across different clocks and supports serialization to persist scoring data
+   across restarts (#1146).
+ * `Event::PaymentSent` now includes the full fee which was spent across all
+   payment paths which were fulfilled or pending when the payment was fulfilled
+   (#1142).
+ * `NetGraphMsgHandler` now takes a `Deref` to the `NetworkGraph`, allowing for
+   shared references to the graph data to make serialization and references to
+   the graph data in the `InvoicePayer`'s `Router` simpler (#1149).
+ * `routing::Score::channel_penalty_msat` has been updated to provide the
+   `NodeId` of both the source and destination nodes of a channel (#1133).
```
Comment (ariard) [inline CHANGELOG.md:36]: I don't see a mention for #1104, IIUC it's a lightweight behavior change in case of crash ?

**Review by ariard — COMMENTED** [review]
Otherwise SGTM. I checked the change logs since 0.0.102 and that all relevant new stuff is mentioned.

**Comment by ariard** [comment]
ACK b43e814

**Review by valentinewallace — APPROVED** [review]

**Comment by TheBlueMatt** [comment]
Squashed without diff. Will land after CI :tada: 

```
$ git diff-tree -U1 b43e81402 2b837bb27
$
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,69 @@
...
+ * This release is almost entirely focused on a new API in the
+   `lightning-invoice` crate - the `InvoicePayer`. `InvoicePayer` is a
+   struct which takes a reference to a `ChannelManager` and a `NetworkGraph`
+   and retries payments as paths fail. It limits retries to a configurable
+   number, but is not serialized to disk and may retry additional times across
+   a serialization/load. In order to learn about failed payments, it must
+   receive `Event`s directly from the `ChannelManager`, wrapping a
+   user-provided `EventHandler` which it provides all unhandled events to
+   (#1059).
+ * `get_route` has been renamed `find_route` (#1059) and now takes a `Payee`
```
Comment (jkczyz) [inline CHANGELOG.md:13]: `Payee` is actually inside `RouteParameters`, which `find_route` takes.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,69 @@
...
+   struct in replacement of a number of its long list of arguments (#1134).
+   `Payee` is further stored in the `Route` object returned and provided in the
+   `RouteParameters` contained in `Event::PaymentPathFailed` (#1059).
+ * `ChannelMonitor`s must now be persisted after calls which provide new block
```
Comment (jkczyz) [inline CHANGELOG.md:17]: s/which/that

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,69 @@
...
+   The `Persist` API has been updated to `Option`ally take the
+   `ChannelMonitorUpdate` as persistence events that result from chain data no
+   longer have a corresponding update (#1108).
+ * `routing::Score` now has a `payment_path_failed` method which it can use to
+   learn which channels often fail payments. It is automatically called by
+   `InvoicePayer` for failed payment paths (#1144).
+ * The default `Scorer` implementation is now a type alias to a type generic
+   across different clocks and supports serialization to persist scoring data
+   across restarts (#1146).
+ * `Event::PaymentSent` now includes the full fee which was spent across all
```
Comment (jkczyz) [inline CHANGELOG.md:29]: May also want to mention that this event and `PaymentPathFailed` now include the `PaymentId`.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,69 @@
...
+ * Delay disconnecting peers if we receive messages from them even if it takes
+   a while to receive a pong from them. Further, avoid sending too many gossip
+   messages between pings to ensure we should always receive pongs in a timely
+   manner. Together, these should significantly reduce instances of us failing
+   to remain connected to a peer during initial gossip sync (#1137).
+ * If a payment is sent, creating an outbound HTLC and sending it to our
+   counterparty (implying the `ChannelMonitor` was persisted on disk), but the
+   `ChannelManager` was not persisted prior to shutdown/crash, no
+   `Event::PaymentPathFailed` event will be generated if the HTLC is eventually
+   failed on chain (#1104).
```
Comment (jkczyz) [inline CHANGELOG.md:48]: Seems to be worded in terms of what the bug was whereas the previous bullet is worded in terms of how the bug was fixed.

**Review by jkczyz — COMMENTED** [review]
Post-merge review of the `CHANGELOG`. All else LGTM! 🎉

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,64 @@
+# 0.0.103 - 2021-11-03
+
+## API Updates
+ * This release is almost entirely focused on a new API in the
+   `lightning-invoice` crate - the `InvoicePayer`. `InvoicePayer` is a
+   struct which takes a reference to a `ChannelManager` and a `NetworkGraph`
```
[the remark this answers is not in the document]
Comment (TheBlueMatt) [reply CHANGELOG.md:6]: Probably, IMO.

File: CHANGELOG.md (hunk 4 above)
Comment (TheBlueMatt) [reply CHANGELOG.md:29]: Which PR # was that?

Comment (jkczyz) [reply CHANGELOG.md:29]: Same PR (#1059).

**Pull request merged**